### PR TITLE
Closes #134: Widget drag and drop

### LINKED ISSUE #134: Ability to drag and drop widgets
**Is your feature request related to a problem? Please describe.**
As a user, I want to be able move widgets around in my journal. This allows more customisability of the journal.

**Describe the solution you'd like**
Implement a way to drop and drop widgets inside the journal.

**Describe alternatives you've considered**
Manually entering positions/coordinates on the screen. Drag and drop is an easy to use and intuitive solution.

**Additional context**
The locations of the widgets should be relative to the screen size ideally. Positions should adjust accordingly when switching to a different screen size. Maybe use some sort of grid.

Blocked by #135. Depends on the journal view of all the widgets (#50, #61, etc)

### PR DESCRIPTION
**Pull request by ambiyu**
Closes #134

## Proposed Changes

- Implemented widget drag and drop
- Currently the drag and drop area is for the entire journal - will likely limit to inside the pages only in the future.
- Can add/remove widgets by changing the widgets array in the Journal component
- Positions (top, left) of the widget are currently not accurate. Likely associated with #149. Once this is resolved, it would be good to implement some sort of grid system to scale with screen size.

### CODE REVIEW

File: telos-frontend/src/views/journal/Journal.jsx [hunk 1]
```diff
@@ -1,15 +1,65 @@
...
+import update from 'immutability-helper';
 import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
 import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import DraggableWidget from '../../dnd/DraggableWidget';
+import { WidgetTypes } from '../../dnd/WidgetTypes';
 import Page from './Page';
+import './Journal.css';
+
+const Journal = () => {
+  const [widgets, setWidgets] = useState([
```
Comment (RawiriHohepa) [inline telos-frontend/src/views/journal/Journal.jsx:42]: The widgets state should be local to a single page component - are you able to move the widget functionality to within Page.jsx?

File: telos-frontend/src/views/journal/Journal.jsx [hunk 2]
```diff
@@ -1,15 +1,65 @@
...
+        WidgetTypes.CALENDAR,
+        WidgetTypes.HABIT_TRACKER,
+        WidgetTypes.TEXT,
+      ],
+      drop(item, monitor) {
+        const delta = monitor.getDifferenceFromInitialOffset();
+
+        const left = Math.round(item.left + delta.x);
+        const top = Math.round(item.top + delta.y);
+        console.log(left, top);
```
Comment (RawiriHohepa) [inline telos-frontend/src/views/journal/Journal.jsx]: What's the reasoning for keeping this console log in?

**Review by RawiriHohepa — APPROVED** [review]
Great job on getting this core functionality working!! Just a couple minor things to fix up

File: telos-frontend/src/views/journal/Journal.jsx (hunk 2 above)
Comment (ambiyu) [reply telos-frontend/src/views/journal/Journal.jsx]: Something I forgot to remove. I will remove that.

**Review by lhuijser — APPROVED** [review]
Looks amazing! Great job on this :) I agree with Rawiri's comments but other than that I am happy to approve

File: telos-frontend/src/views/journal/Journal.jsx (hunk 1 above)
Comment (ambiyu) [reply telos-frontend/src/views/journal/Journal.jsx:42]: This is just the first PR for the basic drag and drop functionality. There will be a second PR to add this widget state into the page component. I need to set the drag/drop container right for the Page component but there are currently issues with the css and styling (#155, #149). I would like these to be fixed before I add it onto the page component as I don't want to run into any issues after.

Comment (RawiriHohepa) [reply telos-frontend/src/views/journal/Journal.jsx:42]: Sounds good, do we have an issue created that we can link as being blocked by #134 or will it be created later?

Comment (ambiyu) [reply telos-frontend/src/views/journal/Journal.jsx:42]: I will be creating a new issue

Comment (RawiriHohepa) [reply telos-frontend/src/views/journal/Journal.jsx:42]: Cool, feel free to merge :)

**Pull request merged**